### PR TITLE
Backports and pin down dependencies (v2-edge)

### PR DESCRIPTION
**Pull request by roosterfish**
Backports the entire delta from `main` into `v2-edge`.

Also pins down the Micro* deps for docs, pipeline and go.

Closes https://github.com/canonical/microcloud/issues/374.

### CODE REVIEW

**Comment by roosterfish** [comment]
I have now backported that last minute doc changes into this PR. The test run before was already green.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -148,10 +148,10 @@ jobs:
         # Set of versions to use for the matrix tests.
         os: ["24.04"]
         go: ["1.22.x"]
-        microceph: ["latest/edge"]
-        microovn: ["latest/edge"]
-        lxd: ["5.21/edge"]
```
Comment (masnax) [inline .github/workflows/tests.yml:105]: While we should have tests using the stable track, I think we should also keep testing against `/edge` to catch regressions early.

**Review by masnax — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (roosterfish) [reply .github/workflows/tests.yml:105]: I think it would be great if we run the same set of tests using the `x/edge` and `x/candidate` channels. 
Maybe let's do this as part of moving to the Canonical runners? It will be a lot of long running jobs.

Comment (roosterfish) [reply .github/workflows/tests.yml:105]: Or run them on a daily/weekly schedule.

Comment (roosterfish) [reply .github/workflows/tests.yml:105]: https://github.com/canonical/microcloud/issues/493
